pkg/api/dashboard.go

```diff
@@ -660,25 +661,52 @@ func (hs *HTTPServer) GetDashboardVersions(c *contextmodel.ReqContext) response.
 		Start:        c.QueryInt("start"),
 	}
 
-	res, err := hs.dashboardVersionService.List(c.Req.Context(), &query)
+	versions, err := hs.dashboardVersionService.List(c.Req.Context(), &query)
 	if err != nil {
 		return response.Error(404, fmt.Sprintf("No versions found for dashboardId %d", dash.ID), err)
 	}
 
-	for _, version := range res {
+	const cacheKey = "userLogin-"
+	res := make([]dashver.DashboardVersionMeta, 0, len(versions))
+	for _, version := range versions {
+		msg := version.Message
 		if version.RestoredFrom == version.Version {
-			version.Message = "Initial save (created by migration)"
-			continue
+			msg = "Initial save (created by migration)"
 		}
 
 		if version.RestoredFrom > 0 {
-			version.Message = fmt.Sprintf("Restored from version %d", version.RestoredFrom)
-			continue
+			msg = fmt.Sprintf("Restored from version %d", version.RestoredFrom)
 		}
 
 		if version.ParentVersion == 0 {
-			version.Message = "Initial save"
+			msg = "Initial save"
 		}
+
+		creator := anonString
+		if version.CreatedBy > 0 {
+			key := fmt.Sprintf("%s%d", cacheKey, version.CreatedBy)
+			if cached, found := hs.CacheService.Get(key); found {
+				creator = cached.(string)
+			} else {
+				creator = hs.getUserLogin(c.Req.Context(), version.CreatedBy)
+				if creator != anonString {
+					hs.CacheService.Set(key, creator, time.Second*5)
+				}
+			}
```

Comment:
agree with most of them except:
> I also think we should cache even if the creator is "anonymous".

because the function returns `Anonymous` for any error case:
https://github.com/grafana/grafana/blob/54d7e95be19d5f0cbd3c7d697966b27bd8feac20/pkg/api/dashboard.go#L261-L268

Reply:
:'( gotcha, we should probably change _that_ at some point, but good catch